syncer/peers.go

```diff
@@ -0,0 +1,73 @@
+package syncer
+
+import (
+	"math/big"
+	"sync"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+type NoForkPeer struct {
+	// identifier
+	ID peer.ID
+	// peer's latest block number
+	Number uint64
+	// peer's distance
+	Distance *big.Int
+}
+
+func (p *NoForkPeer) IsBetter(t *NoForkPeer) bool {
+	if p.Number != t.Number {
+		return p.Number > t.Number
+	}
+
+	return p.Distance.Cmp(t.Distance) < 0
+}
+
+type PeerMap struct {
+	sync.Map
+}
+
+func NewPeerMap(peers []*NoForkPeer) *PeerMap {
+	peerMap := new(PeerMap)
+
+	peerMap.PutPeers(peers)
+
+	return peerMap
+}
+
+func (m *PeerMap) PutPeers(peers []*NoForkPeer) {
+	for _, p := range peers {
+		m.Put(p)
+	}
+}
+
+func (m *PeerMap) Put(peer *NoForkPeer) {
```

Comment:
You can make this a variadic argument, like: `Put(peers ...*NoForkPeer)`, use the for loop like in `PutPeers` and have a single function for putting peers into a peer map.

Reply:
Good suggestion. Fixed at 812f2d68d6d0cdc6297b55bbc5f1328e5b05f3ee